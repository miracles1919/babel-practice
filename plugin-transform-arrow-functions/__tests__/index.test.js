const fs = require('fs');
const path = require('path');
const pluginTester = require('babel-plugin-tester').default;

const PluginTransformArrowFunctions = require('../plugin');

const source = fs.readFileSync(path.join(__dirname, '../source.js')).toString();

pluginTester({
  plugin: PluginTransformArrowFunctions,
  pluginName: 'PluginTransformArrowFunctions',
  babelOptions: {
    parserOpts: {
      sourceType: 'unambiguous',
    },
  },
  tests: {
    default: {
      code: source,
      snapshot: true,
    },
  },
});
