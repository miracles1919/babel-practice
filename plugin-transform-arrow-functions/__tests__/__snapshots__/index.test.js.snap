// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginTransformArrowFunctions default: default 1`] = `

function App() {
  this.name = 'app';
}

var a = 1;
var b = 2;
const foo = () => {
  console.log('a', this.a);

  var b = 3;
  const bar = () => {
    console.log('b', this.b);
  };

  bar();
};

const fun = () => {
  console.log(this);
};

var obj = {
  name: 'obj',
  say: () => {
    console.log(this.name);
  },
};

var obj2 = {
  name: 'obj2',
  say: function () {
    return () => console.log(this.name);
  },
};

      ↓ ↓ ↓ ↓ ↓ ↓

var _this = this,
  _this2 = this,
  _this3 = this;

function App() {
  this.name = "app";
}

var a = 1;
var b = 2;

const foo = function () {
  console.log("a", _this.a);
  var b = 3;

  const bar = () => {
    console.log("b", _this.b);
  };

  bar();
};

const fun = function () {
  console.log(_this2);
};

var obj = {
  name: "obj",
  say: function () {
    console.log(_this3.name);
  },
};
var obj2 = {
  name: "obj2",
  say: function () {
    var _this4 = this;

    return function () {
      console.log(_this4.name);
    };
  },
};


`;
